core/src/saros/editor/IFollowModeListener.java

```diff
@@ -27,7 +28,23 @@
     FOLLOWER_SWITCHES_FOLLOWEE
   }
 
-  public void stoppedFollowing(Reason reason);
+  /** Gets called when the <b>local</b> user starts following another session user. */
+  public default void startedFollowing(User target) {
+    // NOP
+  }
+
+  /** Gets called when the <b>local</b> user stops following another session user. */
+  public default void stoppedFollowing(Reason reason) {
+    // NOP;
+  }
 
-  public void startedFollowing(User target);
+  /** Gets called when a <b>remote</b> user starts following another session user. */
+  public default void startedFollowing(final User follower, final User followee) {
```

Comment:
It seems a bit weird that, with the current usage, this information (like who is now following whom) is never used. Instead, the whole state is just queried and the entire information is reloaded. But I guess that goes back to the same reasoning you mentioned in the review today, it being to cumbersome to update specific elements (and just reloading everything instead does not create that much of an overhead).

Reply:
The whole Eclipse IDE can be created on the fly and disposed on the fly. So there is no chance to maintain the "state machine" somewhere in the GUI logic. Of course I only could load a snapshot on then do my updates, but TBH why should a developer does that when such "wonderful" methods are provided unless they would run in O(n^x)  with x >= 2 ?